lib/io/adc.cpp

```diff
@@ -5,45 +5,49 @@
 
 namespace hyped::io {
 
-Adc::Adc(const uint32_t pin) : pin_(pin)
+Adc::Adc(const uint32_t pin, hyped::core::ILogger &logger) : pin_(pin), logger_(logger)
 {
   char buf[100];
   snprintf(buf, sizeof(buf), "/sys/bus/iio/devices/iio:device0/in_voltage%i_raw", pin_);
   file_ = open(buf, O_RDONLY);
-  if (file_ < 0) { /* TODO: log that there is a problem reading raw voltage for pin_*/; }
+  if (file_ < 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to open ADC file");
+  }
 }
 
 Adc::~Adc()
 {
   close(file_);
 }
 
-std::optional<uint16_t> Adc::read()
+std::optional<uint16_t> Adc::readValue()
 {
   const std::optional<uint16_t> raw_voltage = resetAndRead4(file_);
   if (raw_voltage) {
-    /* TODO: log the value */
+    logger_.log(hyped::core::LogLevel::kDebug, "Raw voltage: %i", raw_voltage.value());
     return *raw_voltage;
   }
   return std::nullopt;
 }
 
-std::optional<uint16_t> resetAndRead4(const int file_descriptor)
+std::optional<uint16_t> Adc::resetAndRead4(const int file_descriptor)
 {
-  const auto offset = lseek(file_descriptor, 0, SEEK_SET);        // reset file pointer
+  const auto offset = lseek(file_descriptor, 0, SEEK_SET);  // reset file pointer
   if (offset != 0) {
-    /*TODO: log that we failed in resetting file_descriptor's file offset*/
+    logger_.log(hyped::core::LogLevel::kInfo, "Failed to reset file offset");
```

Comment:
How are we differentiating between info and fatal errors? This seems just as bad as not being able to open the ADC file.